Add `/roadmap/dencun/` page (#12413)
ethereum/ethereum-org-website: public/content/roadmap/dencun/index.md

---
title: Cancun-Deneb (Dencun) FAQ
description: Frequently asked questions regarding the Cancun-Deneb (Dencun) network upgrade
lang: en
---

# Cancun-Deneb (Dencun) {#dencun}

Cancun-Deneb (Dencun) is an upgrade to the Ethereum network which activates **Proto-Danksharding (EIP-4844)**, introducing temporary data **blobs** for cheaper layer 2 (L2) rollup storage.

A new transaction type enables rollup providers to store data in a more cost-effective manner, in what are known as "blobs". These are guaranteed to be made available to the network for a few weeks, or more precisely, 4096 [epochs](/glossary/#epoch). After this period, blobs are pruned from the network but applications can still verify the validity of their data using proofs. 

This significantly reduces the cost of using rollups, limits chain growth, and helps to support more users while maintaining security and a decentralized set of node operators.

## When do we expect that rollups will reflect lower fees as a result of Proto-Danksharding? {#when}

> - This upgrade is set for activation at the start of epoch 269568, occurring on **13-Mar-2024 (UTC)**

Reviewer: ```suggestion
- This upgrade actives at epoch 269568, on **13-Mar-2024 (UTC)**
```

- All major rollup providers, such as Arbitrum or Optimism, have signaled that blobs will be supported immediately following the upgrade
- Individual rollup support may vary as each rollup must upgrade to take advantage of the new blob space

## How can ETH be converted after the hard fork? {#scam-alert}

> - **There is NO need to convert your ETH**. Your account balances will not change, and the ETH you hold will continue to be available in the same form after the upgrade (hard fork).

Reviewer: to catch seo

```suggestion
- **There is NO need to convert your ETH** after Ethereum's Dencun upgrade. Your account balances will not change, and the ETH you hold will continue to be available in the same form after the upgrade (hard fork).
```

- L1 gas reduction will be proportional to adoption/usage of blob data by rollup providers
- L1 gas is likely to remain competitive from non-rollup related activity
- Rollups that adopt the use of blob space will demand less L1 gas, helping push L1 gas fees downward in the near-term
- Blob space is still limited, so if blobs within a block are saturated/full, then rollups may be required to post their data as permanent data in the meantime, which would drive L1 and L2 gas prices up

## Will this reduce fees on other EVM layer 1 blockchains? {#alt-l1-fee-impact}

No. The benefits of Proto-Danksharding are specific to Ethereum layer 2 rollups that store their proofs on layer 1 (Mainnet).

Simply being compatible with the Ethereum Virtual Machine (EVM) does not mean that a network will see any benefit from this upgrade. Networks that operate independently of Ethereum (whether EVM compatible or not) do not store their data on Ethereum and will not see any benefit from this upgrade.

[More about layer 2 rollups](/layer-2/)

## More of a visual learner? {#visual-learner}

<YouTube id="HT9PHWloIiU" />

> _Finematics: Unlocking Ethereum's Scaling, EIP-4844_

Reviewer: we typically do 'title — source' in other docs

```suggestion
_Unlocking Ethereum's Scaling, EIP-4844 — Finematics _
```

<YouTube id="dFjyUY3e53Q" />

> _Bankless: Blobspace 101 with Domothy_

Reviewer: ```suggestion
_Blobspace 101 with Domothy — Bankless_
```

---
title: Cancun-Deneb (Dencun) FAQ
description: Frequently asked questions regarding the Cancun-Deneb (Dencun) network upgrade
lang: en
---

# Cancun-Deneb (Dencun) {#dencun}

Cancun-Deneb (Dencun) is an upgrade to the Ethereum network which activates **Proto-Danksharding (EIP-4844)**, introducing data **blobs** for cheaper layer 2 (L2) rollup storage.

A new transaction type enables rollup providers to store data in a more cost-effective manner, in what are known as "blobs", which are guaranteed to be made available to the network, but can be discarded after a couple weeks.

This significantly reduces the cost of using rollups, limits chain growth, and helps to support more users while maintaining security and a decentralized set of node operators.

## When do we expect that rollups will reflect lower fees as a result of Proto-Danksharding? {#when}

> - This upgrade is set for activation at start of [epoch](/glossary/#epoch) 269568, occurring on **13-Mar-2024 (UTC)**

Reviewer 2: maybe date (13-Mar-2024) should be formatted according to locale

Author: Yeah, agree... this isn't really possible without some JavaScript so I put up #12426 to add a custom component that maps to the `time` html element to dynamically render a markdown date according to `locale`

---
title: Cancun-Deneb (Dencun) FAQ
description: Frequently asked questions regarding the Cancun-Deneb (Dencun) network upgrade
lang: en
---

# Cancun-Deneb (Dencun) {#dencun}

> Cancun-Deneb (Dencun) is an upgrade to the Ethereum network which activates **Proto-Danksharding (EIP-4844)**, introducing data **blobs** for cheaper layer 2 (L2) rollup storage.

Reviewer 3: ```suggestion
Cancun-Deneb (Dencun) is an upgrade to the Ethereum network which activates **Proto-Danksharding (EIP-4844)**, introducing data **blobs** for cheaper [layer 2 (L2)](/layer-2/) rollup storage.
```

Author: I have a link to this a bit further down to avoid taking users away during the intro

Reviewer 3: ok then maybe use glossary here?

Author: Sounds good, just added that in

> Cancun-Deneb (Dencun) is an upgrade to the Ethereum network which activates **Proto-Danksharding (EIP-4844)**, introducing data **blobs** for cheaper layer 2 (L2) rollup storage.

Reviewer 4: ```suggestion
Cancun-Deneb (Dencun) is an upgrade to the Ethereum network which activates **Proto-Danksharding (EIP-4844)**, introducing ephemeral data **blobs** for cheaper layer 2 (L2) rollup storage.
```

to highlight their unique property

Author: Only push back here is assuming readers will know what ephemeral is; Was looking to use a more descriptive word here

> A new transaction type enables rollup providers to store data in a more cost-effective manner, in what are known as "blobs", which are guaranteed to be made available to the network, but can be discarded after a couple weeks.

Reviewer 4: ```suggestion
A new transaction type enables rollup providers to store data in a more cost-effective manner, in what are known as "blobs". These are guaranteed to be made available to the network for a few weeks, or more precisely, 4096 epochs. After this period, blobs are pruned from the network but applications can still verify the validity of their data using proofs.
```

- All major rollup providers, such as Arbitrum or Optimism, have signaled that blobs will be supported immediately following the upgrade
- Individual rollup support may vary as each rollup must upgrade to take advantage of the new blob space

## How can ETH be converted after the hard fork? {#scam-alert}

- **There is NO need to convert your ETH**. Your account balances will not change, and the ETH you hold will continue to be available in the same form after the upgrade (hard fork).
- <Emoji text="⚠️" /> <strong>Anyone telling you to upgrade your ETH is attempting to scam you!</strong> Users do not need to do anything for this upgrade, and your assets will remain unaffected.

[More on recognizing and avoiding scams](/security/)

## What problem is the Dencun network upgrade solving? {#network-impact}

Dencun primarily addresses **scalability** (handling more users and more transactions) with **affordable fees**, while **maintaining decentralization** of the network.

The Ethereum community has been taking a "rollup-centric" approach to its growth, which places layer 2 rollups as the primary means to safely support more users.

> Rollup networks handle the _processing_ (or "execution") of transactions separate from Mainnet, and then publish a cryptographic proof of the results back to Mainnet for record keeping. Storing these proofs carries an expense (in the form of [gas](/glossary/#gas)), which prior to Proto-Danksharding had to be stored permanently by all network node operators, making it an expensive task.

Reviewer 4: This may be too in the weeds, but ZKRs publish a proof while ORs published compressed transaction data.

> The introduction of Proto-Danksharding in the Dencun upgrade adds cheaper data storage for these proofs by only requiring node operators to store this data for about 18 days, after which is can safely be removed to prevent expansion of hardware requirements.

Reviewer 3: ```suggestion
The introduction of Proto-Danksharding in the Dencun upgrade adds cheaper data storage for these proofs by only requiring node operators to store this data for about 18 days, after which data can be safely removed to prevent expansion of hardware requirements.
```

> The introduction of Proto-Danksharding in the Dencun upgrade adds cheaper data storage for these proofs by only requiring node operators to store this data for about 18 days, after which is can safely be removed to prevent expansion of hardware requirements.

Reviewer 4: Consider adding something like: Because rollups typically have a withdrawal period of 7 days, their security model is unchanged as long as blobs are available on L1 for this duration. The 18 day pruning window provides a significant buffer to this period.

[More on scaling Ethereum](/roadmap/scaling/)

## How is old blob data accessed? {#historical-access}

While regular Ethereum nodes will always hold the _current state_ of the network, historical blob data can be discarded after ~18 days. Before it can be discarded, Ethereum guarantees that this data has been made available to all network participants, to allow time for:

- interested parties to download and store the data
- all rollup challenge periods to be completed

> - and the rollup transactions to be considered finalized.

Reviewer 3: ```suggestion
- and the rollup transactions to be considered finalized
```

Author: [<img width="685" alt="image" src="https://github.com/ethereum/ethereum-org-website/assets/54227730/9fdd7e7f-4922-4164-a50c-87efe6fa04b3">](https://www.microsoft.com/en-us/microsoft-365-life-hacks/writing/punctuating-bullet-point-lists)

think we need to add punctuation to the others

## How does this upgrade contribute to the broader Ethereum roadmap? {#roadmap-impact}

Proto-Danksharding sets the stage for the full implementation of [Danksharding](/roadmap/danksharding/), which splits up the burden of storing rollup data amongst node operators, so each only has to handle a small portion of the total. This will <!-- TIME-SENSITIVE --> enable significantly more data blobs per block, and help Ethereum grow by another order of magnitude.

This is critical to [scale Ethereum to support billions of users](/roadmap/scaling/) with affordable fees and more advanced applications, while maintaining a decentralized network of participants. Without this, hardware requirements for node operators would continue to climb over time, requiring more-and-more expensive machinery to operate, pricing out users who would [otherwise participate](/run-a-node/). This would consolidate node operators to a select few who could afford the hardware, hindering decentralization.

## Does this upgrade affect all Ethereum consensus and validator clients? {#client-impact}

> Yes. Proto-Danksharding (EIP-4844) involves changes to both the execution clients _and_ consensus clients. All production Ethereum clients have released updates to support the upgrade. Node operators must upgrade to a supported version to stay in sync with the chain after the upgrade. <!-- TIME-SENSITIVE --> [See details on supported client releases](https://blog.ethereum.org/2024/02/27/dencun-mainnet-announcement#client-releases)

Reviewer 3: why is the "and" in italic here?

Author: emphasizing the "both" clause

Reviewer 3: i didnt get this, but maybe native speakers do? hmm for me it looked very strange

_Validator_ software is handled by the consensus clients, which have all been updated to accommodate the upgrade.

## How does Cancun-Deneb (Dencun) affect Goerli or other Ethereum testnets? {#testnet-impact}

- Devnets, Goerli, Sepolia and Holesky have all undergone the Dencun upgrade and have Proto-Danksharding fully functioning
- Rollup developers can use these networks for EIP-4844 testing
- Most users will be completely unaffected by this change to each testnet

> ## What is the economic impact for the included Ethereum Improvement Proposals (EIPs)? {#economic-impact}

Reviewer 4: It seems a bit narrow to focus on _only_ potential economic impacts, while there many other dimensions of impact to the upgrade.

### Cancun (Execution) {#economic-impact-cancun}

[EIP-4844](https://eips.ethereum.org/EIPS/eip-4844) - *Shard blob transactions (Proto-Danksharding)*

- Introduction and utilization of blob-space reduces L1 gas demand in the near-term, making all transactions less expensive on average ([see L1 fee impact below](#l1-fee-impact))
- This upgrade primarily affects L2 rollup fees, making them significantly cheaper, but also helps offload some demand for L1 block space
- Lower L1 gas costs means the amount of ETH burned from transaction fees is also reduced, making ETH less deflationary
- Lower gas costs make the network more accessible to new users and new use cases, which may reflexively increase demand for block space over time

> [EIP-7516](https://eips.ethereum.org/EIPS/eip-7516) - *Blob base fee opcode*

Reviewer 4: This is not correct: the fee market is introduced in 4844, 7516 just tells you what the current price is.

- Introduces a new fee market, similar to the EIP-1559 mechanism, dedicated to pricing blob space
- This helps offload demand for traditional L1 block space, reducing fees and reducing the amount of ETH burned on average

### Deneb (Consensus) {#economic-impact-deneb}

[EIP-4844](https://eips.ethereum.org/EIPS/eip-4844) - [See above](#economic-impact-cancun)

[EIP-7044](https://eips.ethereum.org/EIPS/eip-7044) - *Perpetually valid signed voluntary exits*

> - This reduces the trust assumptions for users of pooled staking or staking-as-a-service platforms by providing a self-sovereign means of exiting validator funds without any permission from a third-party node operator

Reviewer 4: ```suggestion
- This reduces the trust assumptions for users of pooled staking or staking-as-a-service platforms by providing a perpetual trustless means of exiting validator funds without any permission from a third-party node operator
```

You can already do "self sovereign" exits, the difference with 7044 is those exit messages no longer expire.

- In the event this encourages more users to stake, the increase in total ETH staked will increase the issuance of new ETH
- In the event users are more likely to exit staking, this will decrease the total ETH staked, reducing the issuance of new ETH

[EIP-7514](https://eips.ethereum.org/EIPS/eip-7514) - *Add max epoch churn limit*

> - This caps the rate at which new ETH can be staked

Reviewer 4: Again, the motivation for this EIP is not to cap issuance but to limit networking overhead for clients, and potentially prevent too large of a % of ETH to be staked before issuance curve changes are more seriously considered.

- This does NOT mean that new validators cannot join, but removes the exponential climb in how many are allowed to join at a time

## Will all transactions on L2s now use temporary blob space, or will you be able to choose? {#calldata-vs-blobs}

Rollup transactions are not required to use blob space for their data (though they are economically incentivized to do so). Each can optionally write their data to blobs (guaranteed available for all challenge periods, very cheap) or to smart contract "calldata" (permanent storage, more expensive). Rollup providers will decide which to use based on the current demand for blob space. If blobs are in high demand, the rollup may decide to use calldata to ensure the data is posted in a timely manner.

Although this decision is likely to be performed behind-the-scenes by rollup providers, it is theoretically possible for a user to choose which storage type to use. This would require the rollup provider to expose this option to the user, which introduces additional complexities when bundling transactions in a cost-effective manner. See individual rollup provider documentation for more details.

> ## Would this reduce layer 1 (L1) gas fees in any way? {#l1-fee-impact}

Reviewer: Will 4844 reduce L1 gas

> Yes, but significantly less than on layer 2 rollups. A new gas market is introduced exclusively for blob space, for use by rollup providers. This reduces the demand from these rollups for L1 block space, pushing gas prices downward.

Reviewer: I think this should be phrased as more "no but maybe..."

- the upgrade does not affect the gas needed for an L1 transaction
- but L2 transaction do have an impact on gas of L1
- therefore if L2 transactions goes down, we might see a small reduction in fees

Author: An L2 that switches to using blob space is no longer consuming execution block space. This will reduce the burden for everyone else on L1.

Reviewer: I just think we should find a more nuanced way of saying "yes... kinda" because it's only true (and to a hard to measure, potentially tiny amount?) if we're standing still.

> Yes, but significantly less than on layer 2 rollups.

what is significantly less?

_Although fees on L1 may be reduced by off-loading rollup data to blobs, this upgrade primarily focuses on the reduction of L2 fees. Reduction of fees on L1 (Mainnet) may occur as a second-order effect to a lesser extent._

- L1 gas reduction will be proportional to adoption/usage of blob data by rollup providers
- L1 gas is likely to remain competitive from non-rollup related activity
- Rollups that adopt the use of blob space will demand less L1 gas, helping push L1 gas fees downward in the near-term
- Blob space is still limited, so if blobs within a block are saturated/full, then rollups may be required to post their data as permanent data in the meantime, which would drive L1 and L2 gas prices up

## Will this reduce fees on other EVM layer 1 blockchains? {#alt-l1-fee-impact}

No. The benefits of Proto-Danksharding are specific to Ethereum layer 2 rollups that store their proofs on layer 1 (Mainnet).

> Simply being compatible with the Ethereum Virtual Machine (EVM) does not mean that a network will see any benefit from this upgrade. Networks that operate independently of Ethereum (whether EVM compatible or not), such as Polygon PoS, Binance Smart Chain, Solana, Avalanche, or Gnosis Chain, do not store their data on Ethereum and will not see any benefit from this upgrade.

Reviewer: I don't think we need to name specific L1s here

[More about layer 2 rollups](/layer-2/)

## More of a visual learner? {#visual-learner}

> <YouTube id="HT9PHWloIiU" />

Reviewer 3: its a bad practice to put video at the bottom of the page, should be higher, basically almost at the top. 

Also the headline "More of a visual learner?" is redundant

Reviewer 3: Those who would prefer video content likely might not even reach the end of the page. As a user I would like to have the choice how to consume content at top (consume by reading vs consume by watching a video)

Author: The header and position is in alignment with the rest of the repo

Author: Happy to change, just noting this is a pattern previously established

Reviewer: I don't see this as blocking. Agree w/@konopkja that we should reconsider the pattern, but agree w/@wackerow that we're just using existing patterns and this PR might not be the best place to reconsider those.

Reviewer 3: we have already reconsidered that and changed it on few pages during comprehension small edits btw, but agree not a blocker here

_Finematics: Unlocking Ethereum's Scaling, EIP-4844_

<YouTube id="dFjyUY3e53Q" />

_Bankless: Blobspace 101 with Domothy_

## Further reading {#further-reading}

- [EIP4844.com](https://www.eip4844.com/)
- [EIP-4844: Shard blob transactions (Proto-Danksharding)](https://eips.ethereum.org/EIPS/eip-4844)
- [Dencun Mainnet Announcement](https://blog.ethereum.org/2024/02/27/dencun-mainnet-announcement) - _Ethereum Foundation blog_
- [The Hitchhiker's Guide to Ethereum: Proto-Danksharding](https://members.delphidigital.io/reports/the-hitchhikers-guide-to-ethereum/#proto-danksharding-eip-4844) - _Jon Charbonneau_
- [Proto-Danksharding FAQ](https://notes.ethereum.org/@vbuterin/proto_danksharding_faq) - _Vitalik Buterin_

> - [An In-depth Explanation of EIP-4844: The Core of the Cancun Upgrade](https://medium.com/@ebunker.io/an-in-depth-explanation-of-eip-4844-the-core-of-the-cancun-upgrade-de7b13761d2c) - _Ebunker_

Reviewer 4: I'll shill my own piece: https://tim.mirror.xyz/HzH5MpK1dnw7qhBSmzCfdCIxpwpD6DpwlfxtaAwEFro
